pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -941,6 +955,364 @@ class StateVectorCudaMPI
+
+                // allocate an external buffer if needed
+                PL_CUSPARSE_IS_SUCCESS(cusparseSpMV_bufferSize(
+                    /* cusparseHandle_t */ handle,
+                    /* cusparseOperation_t */ CUSPARSE_OPERATION_NON_TRANSPOSE,
+                    /* const void* */ &alpha,
+                    /* cusparseSpMatDescr_t */ mat,
+                    /* cusparseDnVecDescr_t */ vecX,
+                    /* const void* */ &beta,
+                    /* cusparseDnVecDescr_t */ vecY,
+                    /* cudaDataType */ data_type,
+                    /* cusparseSpMVAlg_t */ CUSPARSE_SPMV_ALG_DEFAULT,
+                    /* size_t* */ &bufferSize));
+
+                DataBuffer<cudaDataType_t, int> dBuffer{bufferSize, device_id,
+                                                        stream_id, true};
+
+                // execute SpMV
+                PL_CUSPARSE_IS_SUCCESS(cusparseSpMV(
+                    /* cusparseHandle_t */ handle,
+                    /* cusparseOperation_t */ CUSPARSE_OPERATION_NON_TRANSPOSE,
+                    /* const void* */ &alpha,
+                    /* cusparseSpMatDescr_t */ mat,
+                    /* cusparseDnVecDescr_t */ vecX,
+                    /* const void* */ &beta,
+                    /* cusparseDnVecDescr_t */ vecY,
+                    /* cudaDataType */ data_type,
+                    /* cusparseSpMVAlg_t */ CUSPARSE_SPMV_ALG_DEFAULT,
+                    /* void* */ reinterpret_cast<void *>(dBuffer.getData())));
+
+                // destroy matrix/vector descriptors
+                PL_CUSPARSE_IS_SUCCESS(cusparseDestroySpMat(mat));
+                PL_CUSPARSE_IS_SUCCESS(cusparseDestroyDnVec(vecX));
+                PL_CUSPARSE_IS_SUCCESS(cusparseDestroyDnVec(vecY));
+
+                color = 1;
+            }
+
+            PL_CUDA_IS_SUCCESS(cudaDeviceSynchronize());
+            mpi_manager_.Barrier();
+
+            if (mpi_manager_.getRank() == i) {
+                color = 1;
+            }
+
+            auto new_mpi_manager =
+                mpi_manager_.split(color, mpi_manager_.getRank());
+            int reduce_root_rank = -1;
+
+            if (mpi_manager_.getRank() == i) {
+                reduce_root_rank = new_mpi_manager.getRank();
+            }
+
+            mpi_manager_.Bcast<int>(reduce_root_rank, i);
+
+            if (new_mpi_manager.getComm() != MPI_COMM_NULL) {
+                new_mpi_manager.Reduce<CFP_t>(d_tmp.getData(),
+                                              d_tmp_res.getData(), length_local,
+                                              reduce_root_rank, "sum");
+            }
+        }
+
+        mpi_manager_.Barrier();
+
+        local_expect = innerProdC_CUDA(d_tmp_res.getData(), BaseType::getData(),
+                                       BaseType::getLength(), device_id,
+                                       stream_id, getCublasCaller())
+                           .x;
```

Comment:
Strange formatting, no (the `.x`)?

Reply:
it's strange that it passed format check.